include/range/v3/view/group_by.hpp

```diff
@@ -73,30 +71,32 @@ namespace ranges
             {
                 iterator_t<CRng> first_;
                 semiregular_box_ref_or_val_t<Fun, IsConst> fun_;
+                mutable bool first_invokation_{true};
```

Comment:
Current practice is to not use mutable data members and use non-`const` member functions instead. But that seems like a breaking change.

Reply:
How so? Just changed it, the test passes fine. Will commit now.

Reply:
It might be observable behaviour. But this is a bug fix, so it's justified.

Reply:
Hi @johelegp,

After my last change (to remove the mutable and make the `pred::operator()` non-const) I've faced another issue. In such a code:
```
const auto groups = /*...*/ | views::group_by(/*...*/);
for (const auto group : groups) { // Won't compile! Does only after removing the const.
    // ...
}
```
Would you agree that this is unacceptable and unexpected?
And if so, wouldn't it be an acceptable tradeoff to make that `prad::operator()` const again with a mutable member?

Reply:
I'd say the current state is the right one. Views should be non-`const`.